### DISCUSSION TITLE
push to origin hangs for username.github.io

### DISCUSSION THREAD

**Discussion by ScottAkenhead** [community/community#31072] [Other Feature Feedback, Questions, & Ideas]
GitHub Desktop on Mac (up to date): **push to origin** hangs. Files appear to be transferred, I can watch the progress (see picture below), but the process does not end (spinning icon, forever) and the repo on GitHub.com is not updated. This problem might be limited to pushing to username.github.io, which I am trying to use as a Quarto website/blog. 

<img width="476" alt="image" src="https://user-images.githubusercontent.com/42452064/187049701-26fc9a27-5728-4942-892e-aaf61f94839e.png"> 

Push to origin worked for local repo of Quarto code to create that site, so:  content and sizes are OK.  See for yourself:  [code](https://github.com/ScottAkenhead/blog-salmon-analysis) and not-updated [website](https:ScottAkenhead.github.io).

I am newbie-newbie-new, so  _methodical_ responses appreciated.

**Comment by jfullstackdev** [comment]
how big are those files ? what is the biggest file in there ?

**Reply by ScottAkenhead** [reply]
Thanks for asking! How big? Not big. Can you see for yourself, maybe? Locally, the hidden folder .git is 6.2 Mbytes, largest  file in there is 3 Mbytes.  
I was able to “push to origin” the .git folder for code plus website, total 12 Mbytes. In passing, that initially failed, but success after restarting Mac and GitHub Desktop.
Is it a problem to have a git repo inside a git repo? So there is a repo (a .git folder) for Quarto code (an Rstudio project) which contains the folder for the website rendered from that code, AND a repo for that website folder. Attempting to please the Git gods, I created the local repo for website folder _starting from_ GitHub.com; which happened without issue.

**Reply by airtower-luna** [reply]
> Is it a problem to have a git repo inside a git repo? So there is a repo (a .git folder) for Quarto code (an Rstudio project) which contains the folder for the website rendered from that code, AND a repo for that website folder.

It's possible, but makes things complicated. The nested repository will become a submodule of the enclosing one. Whether that's suitable depends on your project. See [Git Tools - Submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) for details.

**Comment by ScottAkenhead** [comment]
Turns out there is a lot more than Push to post a Quarto Blog Website to a GitHub.io website. [https://quarto.org/docs/publishing/github-pages.html](https://quarto.org/docs/publishing/github-pages.html).  For me, either (a) publishing via GitHub is complicated or (b) Quatro docs for this are needlessly daunting. Dropping  the GitHub approach; tryingQuarto Pub.  [https://quarto.org/docs/publishing/quarto-pub.html](https://quarto.org/docs/publishing/quarto-pub.html)
kthxbai
